File: solutions/049.group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
const groupAnagrams = strs => {
  let map = {}

  for (let i = 0; i < strs.length; i++) {
    let item = strs[i]
    let ref = sortStr(item)

    if (ref in map) {
      map[ref].push(item)
    } else {
      map[ref] = [item]
    }
  }

  return Object.values(map)// arr
};

const sortStr = s => [...s].sort().join('');

File: solutions/049.group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        map = {}

        for s in strs:
            pure = self.get_pure(s)
            map[pure] = map.get(pure, []) + [s]

        ret = []

        for _, v in map.items():
            ret.append(v)

        return ret

    @staticmethod
    def get_pure(s):
        return ''.join(sorted(list(s)))
